cns/restserver/ipam.go

```diff
@@ -113,6 +114,45 @@ func (service *HTTPRestService) MarkIPsAsPending(numberToMark int) (map[string]c
 	return nil, fmt.Errorf("Failed to mark %d IP's as pending, only marked %d IP's", numberToMark, len(pendingReleaseIPs))
 }
 
+// MarkAllPendingProgrammingIpsAsAvailableUntransacted is the function to update pending programming IPs to available
+// when get NC version failed and we don't want to block IP allocation.
+// Note: this func is an untransacted API as the caller will take a Service lock
+func (service *HTTPRestService) MarkAllPendingProgrammingIpsAsAvailableUntransacted() {
```

Comment:
This doesnt look right to me, this defeats the purpose that we will ensure IP is programmed before we assign it to the pod